src/fsharp/TypeChecker.fs

```diff
@@ -9812,7 +9811,7 @@ and CheckRecursiveBindingIds binds =
 and TcLinearExprs bodyChecker cenv env overallTy tpenv isCompExpr expr cont = 
     match expr with 
     | SynExpr.Sequential (sp,true,e1,e2,m) when not isCompExpr ->
-        let e1',_ = TcStmtThatCantBeCtorBody cenv env tpenv e1
+        let e1',_ = TcStmtThatCantBeCtorBody cenv (ClearContext env) tpenv e1
```

Comment:
@dsyme looks like this is the place where the context needs to be removed. but I'm still checking